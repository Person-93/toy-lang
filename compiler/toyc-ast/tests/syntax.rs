use anyhow::Result;
use chumsky::Parser;
use test_util::{assert_debug_snapshot, snapshots};
use toyc_ast::{ast, tokens::TokenIter};

fn main() -> Result<()> {
  snapshots(|text| {
    let tokens = TokenIter::new(text).into_stream();
    assert_debug_snapshot!(ast::parse::file().parse_recovery(tokens));
  })
}
