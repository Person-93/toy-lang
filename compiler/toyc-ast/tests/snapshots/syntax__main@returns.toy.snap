---
source: compiler/toyc-ast/tests/syntax.rs
assertion_line: 8
expression: "ast::parse::file().parse_recovery(tokens)"

---
(
    Some(
        File {
            attrs: [],
            items: [
                Item {
                    attrs: [],
                    vis: None,
                    item_kind: Function(
                        Function {
                            extern_: None,
                            ident: return_str @ 3..13,
                            generic_params: None,
                            params: [],
                            return_type: Some(
                                Ref(
                                    Ident(
                                        str @ 20..23,
                                    ),
                                ),
                            ),
                            body: Some(
                                Expr {
                                    primary: Literal(
                                        StrLit(
                                            "hello world",
                                        ),
                                    ),
                                    args: None,
                                    span: Span {
                                        lo: 25,
                                        hi: 38,
                                    },
                                },
                            ),
                            span: Span {
                                lo: 0,
                                hi: 38,
                            },
                        },
                    ),
                    span: Span {
                        lo: 0,
                        hi: 38,
                    },
                },
                Item {
                    attrs: [],
                    vis: None,
                    item_kind: Function(
                        Function {
                            extern_: None,
                            ident: return_num @ 43..53,
                            generic_params: None,
                            params: [],
                            return_type: Some(
                                Ident(
                                    i8 @ 59..61,
                                ),
                            ),
                            body: Some(
                                Expr {
                                    primary: CodeBlock(
                                        CodeBlock {
                                            statements: [],
                                            trailing: Some(
                                                Expr {
                                                    primary: Literal(
                                                        NumLit(
                                                            NumLit {
                                                                prefix: None,
                                                                val: 42,
                                                                decimal: None,
                                                                ty: None,
                                                                span: Span {
                                                                    lo: 65,
                                                                    hi: 67,
                                                                },
                                                            },
                                                        ),
                                                    ),
                                                    args: None,
                                                    span: Span {
                                                        lo: 65,
                                                        hi: 67,
                                                    },
                                                },
                                            ),
                                            span: Span {
                                                lo: 65,
                                                hi: 67,
                                            },
                                        },
                                    ),
                                    args: None,
                                    span: Span {
                                        lo: 63,
                                        hi: 69,
                                    },
                                },
                            ),
                            span: Span {
                                lo: 40,
                                hi: 69,
                            },
                        },
                    ),
                    span: Span {
                        lo: 40,
                        hi: 69,
                    },
                },
            ],
            span: Span {
                lo: 0,
                hi: 69,
            },
        },
    ),
    [],
)
